File: abc037_c.kt

fun main(args : Array<String>) { 
  val (n, m) = readLine()!!.split(" ").map { x -> x.toInt() }
  val b:MutableList<Long> = readLine()!!.split(" ").map { x -> x.toLong() }.toMutableList()
  (0 .. m - 2).map { x ->
    b[x] = (x+1) * b[x]
  }
  (m-1 .. b.size - m).map { x ->
    b[x] = m * b[x]
  }
  var f = m - 1
  (b.size -m + 1 .. b.size - 1).map { x ->
    b[x] = f * b[x]
    f--
  }
  //println( b )
  println( b.sum() )

}

File: abc037_c.py

m,n = map(int, input().split())
bs  = list( map(int, input().split()) )
size= len(bs)
a   = []
cnt = 1
for i,b in enumerate(bs):
  if i+1 < n:
    a.append( ((i+1), b))
  elif i > size - n:
    a.append( (n - cnt, b) )
    cnt+=1
  else:
    a.append( (n, b) )

print(sum(b*c for b,c in a))

